### ISSUE TITLE
telegraf 1.13.x: net/http: request canceled (Client.Timeout exceeded while awaiting headers)

### ISSUE THREAD

**Issue by itn3rd77** [influxdata/telegraf#6972]
Since an update from telegraf 1.12.6 to telgraf 1.13.1 or 1.13.2 all input plugins using HTTP/HTTPS (e.g. inputs.http_response or inputs.jolokia2_agent) are throwing errors when monitoring the server where telegraf is running. No matter if localhost or the fqdn is defined. 

```
Error in plugin: Unable to gather metrics for http://127.0.0.1:14115/jolokia: Post http://travic:***@127.0.0.1:14115/jolokia/read: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```

When reverting back to telegraf 1.12.6 everything is fine again.

### Relevant telegraf.conf:

```toml
[[inputs.http_response]]
  ## List of urls to query.
  urls = ["https://localhost:8443"]

  ## Optional TLS Config
  insecure_skip_verify = true
```

### System info:
telegraf 1.13.1/1.13.2
Red Hat Enterprise Linux 7.6

### Steps to reproduce:

1. Define one ore more input plugins making use of http/https protocoll to localhost (e.g. inputs.http_response or inputs.jolokia2_agent)
2. After 10 seconds the first error messages appear in the log

```
Error in plugin: Unable to gather metrics for http://127.0.0.1:14115/jolokia: Post http://travic:***@127.0.0.1:14115/jolokia/read: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```

### Expected behavior:

No errors are logs

### Actual behavior:

After 10 seconds the first error messages appear in the log

```
Error in plugin: Unable to gather metrics for http://127.0.0.1:14115/jolokia: Post http://travic:***@127.0.0.1:14115/jolokia/read: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```

**Comment by danielnelson** [comment]
Strange, I'm not sure what the issue might be.  Can you try with this configuration:
```toml
[[inputs.http_response]]
  urls = ["http://www.example.org"]
```

**Comment by itn3rd77** [comment]
That works fine. As long as the URL I try to monitor is not served by the server itself running telegraf I see no errors popping up.

As far as I can tell the problem seems only to raise if telegraf is requesting an URL served by the server on which telegraf is running. I have no glue what could cause this. Little example to better explain the behavior

Lets say I run two servers A and B. Both have telegraf running and an some kind of HTTP service to monitor.

1. Monitoring HTTP service from server A with telegraf on server A results in the errors I posted. The same is true for Monitoring HTTP service from server B with telegraf on server B
2. If I swap the monitoring e.g. monitoring HTTP service from server A with telegraf on server B. There is no problem.

The really strange thing is that the error only raise on telegraf >= 1.13.x.

Has anything changed on the underlying go library. Can I enable tracing to provide you some insights?

**Comment by danielnelson** [comment]
I'm not aware of any changes that would cause this, but you never know.  Are you running Telegraf 1.12 and 1.13 on the exact same machine?  I suppose a packet capture with tcpdump might be helpful:
```
sudo tcpdump -A -s0 -i lo 'port 9273' -w output.pcap
```

**Comment by M0rdecay** [comment]
I will add that now I switched to version 1.13 on RHEL7 for monitoring applications through jolokia2 and do not observe a similar problem when accessing localhost.

**Comment by ykuksenko** [comment]
I am seeing this with outputs across different servers.
```
[outputs.influxdb] When writing to [http://lamp:8086]: Post http://lamp:8086/write?db=telegraf: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
[agent] Error writing to outputs.influxdb: could not write any address
```

**Comment by danielnelson** [comment]
This error message indicates there was a network timeout, but it could have many causes.  Often it means there is a firewall that is DROPing your packets.

**Comment by ykuksenko** [comment]
I tried `curl http://lamp:8086` from the specific server with the error and got a 403 - so it shouldn't be firewall - at least nothing apparent I can think of - there isn't any rate limiting setup.

**Comment by danielnelson** [comment]
This timeout is for the entire request, it is possible the other side is responding but the response was not received in time.

**Comment by LTKH** [comment]
Hello. I have a similar problem. The problem is repeated on dozens of servers. There is a suspicion that golang has been updated and with it the "net / http" module. It affected.

**Comment by tanuj83** [comment]
I am also facing the same issue 

telegraf-1.12.4
influxdb-1.7.9

**Comment by hex2a** [comment]
same issue,

telegraf 1.13.2
influxdb 1.7.10

InfluxDB logs show a `400`HTTP status code for the `/write` endpoint, there is an error in the response (but no timeout):
```
{"error":"unexpected EOF"}
```

**Comment by ssoroka** [comment]
I believe this is resolved as of Telegraf 1.14; related to https://github.com/influxdata/telegraf/pull/7517
